### ISSUE TITLE
Invalid prop 'component' on Icon

### ISSUE THREAD

**Issue by jjdp** [react-native-community/react-native-elements#78]
```
when I specify a component on an Icon like so:

<Icon
  reverse
  raised
  name="chat"
  color="lightblue"
  component={TouchableOpacity}
  onPress={() => console.log('hello')}
/>
```

it gives this warning Failed prop type: Invalid prop 'component' of type 'function' supplied to 'Icon', expected a single ReactElement

**Comment by NikolaBorislavovHristov** [comment]
try to use 'Component' instead of 'component'

**Comment by dabit3** [comment]
There was a pr that was merged to fix this issue. We will be releasing v0.7.0 with this fix soon. thanks!

**Comment by booboothefool** [comment]
I am still getting this issue with 0.7.1.

**Comment by booboothefool** [comment]
@dabit3 I am still getting this issue with 0.8.2.

**Comment by Monte9** [comment]
@booboothefool opening this issue again. Will look into it soon.

**Comment by Monte9** [comment]
Unable to reproduce this with RNE v0.9.6

Can you confirm that it is still an issue for you? @booboothefool @jjdp. 
If yes, then PR #176 should hopefully fix it. 
/cc @max2three

**Comment by cbjs** [comment]
warn still with RNE v0.9.7

change propType of component to func will fix this.

**Comment by max2three** [comment]
@Monte9 Do you want me to reopen the PR that I had closed earlier?

**Comment by SteveGreenley** [comment]
Still an issue with RNE v0.13.0

**Comment by Monte9** [comment]
@SteveGreenley pushed a fix to `v0.14.0` here: https://github.com/react-native-training/react-native-elements/pull/438/commits/8a6318fb9935a404cd9170b6733f49d3e5c47710

This works on v0.14.0

```
import { TouchableOpacity } from 'react-native';
import { Icon } from 'react-native-elements';

<Icon
  component={TouchableOpacity}
  name="youtube-searched-for"
  type="material"
  size={30}
  color="black"
  onPress={() => console.log('Search Pressed')}
/>
```

**Comment by SteveGreenley** [comment]
Thank you @Monte9. It's working nicely.
